__tests__/sessionInfo/SessionInfoManager.spec.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * Test for SessionCreator
+ */
+import "reflect-metadata";
+import { UuidGeneratorMock } from "../../src/util/__mocks__/UuidGenerator";
+import { AuthenticatedFetcherMock } from "../../src/authenticatedFetch/__mocks__/AuthenticatedFetcher";
+import { LogoutHandlerMock } from "../../src/logout/__mocks__/LogoutHandler";
+import { EmptyStorageUtilityMock } from "../../src/storage/__mocks__/StorageUtility";
+import SessionInfoManager from "../../src/sessionInfo/SessionInfoManager";
+
+describe("SessionInfoManager", () => {
+  const defaultMocks = {
+    uuidGenerator: UuidGeneratorMock,
+    authenticatedFetcher: AuthenticatedFetcherMock,
+    logoutHandler: LogoutHandlerMock,
+    storageUtility: EmptyStorageUtilityMock
+  };
+  function getSessionInfoManager(
+    mocks: Partial<typeof defaultMocks> = defaultMocks
+  ): SessionInfoManager {
+    const sessionManager = new SessionInfoManager(
+      mocks.storageUtility ?? defaultMocks.storageUtility
+    );
+    return sessionManager;
+  }
+
+  describe("update", () => {
+    it("is not implemented yet", async () => {
+      const sessionManager = getSessionInfoManager({
+        storageUtility: EmptyStorageUtilityMock
+      });
+      expect(
+        async () => await sessionManager.update("commanderCool", {})
+      ).rejects.toThrow("Not Implemented");
+    });
+  });
+
+  describe("get", () => {
+    it("creates a session from localstorage", async () => {
+      const storageUtility = defaultMocks.storageUtility;
+      storageUtility.getForUser
+        .mockReturnValueOnce(
+          Promise.resolve("https://zoomies.com/commanderCool#me")
+        )
+        .mockReturnValueOnce(Promise.resolve("true"));
+      const sessionManager = getSessionInfoManager({ storageUtility });
+      const session = await sessionManager.get("commanderCool");
```

Comment:
I don't see where "commanderCool" is being used as the key to **_set_** the session instance that this call to '.get()' is retrieving...?

Reply:
It is not: the storage is simply mocked to return a WebId when asked. The test verifies that if we get a session based on a certain key, the obtained sessionInfo is built based on both the retrieved information (here, the WebID) and the used sessionId.